taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -200,11 +200,18 @@ func (self *Contract) CommitCall(readStorage Reader) {
 }
 
 // Fills contract based on genesis values
-func (self *Contract) ApplyGenesis() error {
+func (self *Contract) ApplyGenesis(get_account func(*common.Address) vm.StateAccount) error {
 	self.lazy_init()
 
-	for _, entry := range self.cfg.GenesisState {
-		self.apply_genesis_entry(&entry.Benefactor, entry.Transfers)
+	make_context := func(caller *common.Address, value *big.Int) (ctx vm.CallFrame) {
+		ctx.CallerAccount = get_account(caller)
+		ctx.Account = get_account(contract_address)
+		ctx.Value = value
+		return
+	}
+
+	for _, entry := range self.cfg.InitialValidators {
+		self.apply_genesis_entry(&entry, make_context)
```

Comment:
you dont need `apply_genesis_entry` function anymore. Those 5 lines from it can be directly here...